Add json definition for generic design system components.

MayaKirova opened the pull request:
Closes #19

MayaKirova committed:
Add json definition for generic avatar.

dkamburov committed:
Add json definition for generic icon.

MayaKirova committed:
Add json definition for generic calendar.

StefanIvanov commented on design model/components.model.json:
```diff
@@ -68,6 +68,50 @@
+            "name": "Calendar",
+            "properties": [
+                {
+                    "name": "Type",
```
```suggestion
                    "name": "Layout",
```

MayaKirova committed:
chore(*): Apply comments.

dkamburov committed:
Apply some review notes

MayaKirova committed:
Add json definition for generic button.

MayaKirova committed:
Add json definition for generic checkbox.

dkamburov committed:
Add json definition for generic date picker

skrustev committed:
Add definitions for Badge, Radio and Switch.

StefanIvanov commented on design model/components.model.json:
```diff
@@ -177,6 +177,47 @@
+                        "default": "Default",
+                        "actionbuttons": "Actionbuttons"
+                    }
+                },
```
I feel that this might not be necessary since the buttons come from the mode if I remember correctly

dkamburov replied:
I agree and am removing it.

StefanIvanov commented on design model/components.model.json:
```diff
@@ -218,6 +218,99 @@
+                    "value": "Round"
+                },
+                {
+                    "name": "Type",
```
Again we may prefer to call it "ContentType"

StefanIvanov replied:
And if the avatar uses Square and Circle we should follow the same convention here

MayaKirova committed:
Add generic Card definition.

MayaKirova committed:
Merge from base.

MayaKirova committed:
Add description for input group.

skrustev committed:
Add definitions for list.

MayaKirova committed:
Add initial grid definition.

StefanIvanov commented on design model/components.model.json:
```diff
@@ -461,6 +461,35 @@
+              "value": "Comfortable"
+            },
+            {
+              "name": "Features",
```
In the UI Kits, we have some features on the grid level that live as separate components that you drag and drop on the grid, like paging, and some features are nested as dummy symbols in the column header such as sorting because they are switched on a per column basis. Is this holding both types of features?

MayaKirova replied:
I imagine this would accumulate all features enabled for the grid, with some additional information on the feature (if such information can be extracted).
Like for example: 
Features: [{
"name": "Filtering",
"mode": "excel"
}, {
"name": "Sorting",
"mode": "single"
}
]
If the feature is enabled just for some of the columns but not for all (for example filtering icon is shown just for 1 of the columns) then that information would be stored in the Columns.  For example:
Column: [{
"name": "Column 1",
"type": "string",
"filterable": true
}]

StefanIvanov replied:
ОК so if under features it is valid for all grid columns in case it is a per-column feature, and if in column X, it is valid only for column X

MayaKirova committed:
Add Dialog.

StefanIvanov commented on design model/components.model.json:
```diff
@@ -0,0 +1,551 @@
+                            "value": ""
+                        },
+                        {
+                            "name": "Type",
```
We should rename this to "ContentType" and use the Type property for line, box, border

StefanIvanov commented on design model/components.model.json:
```diff
@@ -0,0 +1,551 @@
+                    "name": "ContentType",
+                    "valueType": "enum",
+                    "enumType": {
+                        "text": "Text",
+                        "icon": "Icon"
```
I think we should also support icon+text and text+icon

MayaKirova replied:
The current sketch button does not seem to have a button+icon mode. Will such a mode be added or will the button be considered a composite component - something that can be decomposed and templated?

StefanIvanov replied:
I plan to add these "modes" which are actually content types/templates but am not sure in what way they will appear in the definition in order to create the sketch structure

dkamburov committed:
remove the Type of datepicker as it is not needed

MayaKirova committed:
Apply review changes for Button and InputGroup

MayaKirova committed:
Merge branch 'design-system-model-json' of https://github.com/IgniteUI/igniteui-angular-ui-kits.git

MayaKirova committed:
chore(*): Fix prop name.

MayaKirova committed:
chore(*): Describe composite containers as children.

StefanIvanov commented:
@MayaKirova could you also add a TimePicker definition akin to the DatePicker one?

MayaKirova committed:
chore(*): Define RadioGroup component with Radio component child.

dkamburov committed:
Add disabled property for the button and format doc

MayaKirova committed:
Adding TimePicker definition.

MayaKirova committed:
Merge branch 'design-system-model-json' of https://github.com/IgniteUI/igniteui-angular-ui-kits.git

MayaKirova committed:
chore(*): Update dialog definition.

MayaKirova committed:
chore(*): Fix prop name.

MayaKirova committed:
chore(*): Move "disabled" property to RadioGroup level.

MayaKirova committed:
Update Calendar enumTypes.

StefanIvanov commented on design model/components.model.json:
```diff
@@ -0,0 +1,642 @@
+                        "date": "Date",
+                        "phone": "Phone"
+                    }
+                }
```
We may want to add a "disabled" property akin to other form elements

MayaKirova committed:
chore(*): Update Badge and Input properties.

MayaKirova committed:
chore(*): Rename value to defaultValue.

MayaKirova committed:
Rename template container definitions to re-use the same collection as direct child component definitions.
Both will be contained inside childComponents.

MayaKirova committed:
Update Grid definition with Column definitions and Feature definition.

MayaKirova committed:
Add enabled property to features.

MayaKirova committed:
Split enbaled column features from column state. Fetaures stored in separate collection, state related things stored in the properties.

3phase committed:
Updates DatePicker definition's properties to match w/ Angular component ones

MayaKirova commented on design model/components.model.json:
```diff
@@ -194,18 +194,9 @@
-                        "horizontal": "Horizontal"
-                    },
-                    "defaultValue": "Horizontal"
+                    "name": "Vertical",
```
This should be the consistent with the Calendar property. Does not have to match the Angular component property name.

MayaKirova commented on design model/components.model.json:
```diff
@@ -194,18 +194,9 @@
                     "defaultValue": ""
                 },
                 {
-                    "name": "Months",
```
This is used to define the number of months displayed in the calendar, in order to describe a multi-view datepicker. It should not be removed. If the naming is not clear enough it can be named "MonthViews" instead.

3phase committed:
Restoring months property, as it brings information about the multi layout view & Layout, as it's widely used w the same purpose

MayaKirova commented:
@damyanpetev @wnvko @ChronosSF @skrustev  We are thinking of adding another property to the model to define the features for complex components, like the grid:
https://github.com/IgniteUI/igniteui-angular-ui-kits/pull/18/files#diff-a4729536ce7b2d4046c7cf08f1fe4bafR744
Let me know if it would be ok to add such a property to the component definitions model or if you have some other suggestions on where the feature definitions should live feel free to share them.

MayaKirova committed:
chore(*): Remove Toolbar as it has inconsistent structure in IgniteUI for Angular and needs refactoring.  Moving all simple features under Grid/Column  Properties.

MayaKirova committed:
Merge branch 'design-system-model-json' of https://github.com/IgniteUI/igniteui-angular-ui-kits.git

MayaKirova committed:
chore(*): Add summary to column.

MayaKirova committed:
chore(*): Update prop name.

skrustev committed:
Add IDs to the design definitions to reflect fully the generic app model..

skrustev committed:
Change properties that point to asset to be of type "asset"

MayaKirova committed:
Add list item templatable sections as child components.

MayaKirova committed:
Merge branch 'design-system-model-json' of https://github.com/IgniteUI/igniteui-angular-ui-kits.git

MayaKirova committed:
fix typo.

dkamburov committed:
Add linearbar and circularbar

MayaKirova committed:
chore(*): Simpify container names.

dkamburov committed:
Add definition for Slider

dkamburov committed:
Merge branch 'design-system-model-json' of https://github.com/IgniteUI/igniteui-angular-ui-kits into design-system-model-json

dkamburov committed:
Add Snackbar definition

dkamburov committed:
Add toast definition

dkamburov committed:
Add CategoryChart and BulletGraph definitions

dkamburov committed:
Add LinearGauge definition

dkamburov committed:
Add RadialGauge definition

dkamburov committed:
Add tooltip property for category chart

StefanIvanov commented on design model/components.model.json:
```diff
@@ -0,0 +1,1170 @@
+                        "needle": "Needle",
+                        "triangle": "Triangle",
+                        "trapezoid": "Trapezoid"
+                    }
```
add needle brush akin to the value brush in the bullet graph

StefanIvanov replied:
also add a background color

StefanIvanov replied:
also consider major and minor ticks brush

StefanIvanov commented on design model/components.model.json:
```diff
@@ -0,0 +1,1170 @@
+                    "name": "TargetValueBrush",
+                    "valueType": "string",
+                    "value": ""
+                }
```
add a background color

StefanIvanov commented on design model/components.model.json:
```diff
@@ -0,0 +1,1170 @@
+                        "needle": "Needle",
+                        "triangle": "Triangle",
+                        "trapezoid": "Trapezoid"
+                    }
```
add needle brush, background color, major and minor ticks like for the linear gauge

StefanIvanov commented:
@dkamburov I have left some comments for you on the data visualizations

dkamburov committed:
Apply review notes

dkamburov committed:
Add Navdrawer definitions

dkamburov committed:
Add navbar definition

skrustev committed:
Add monthViews property to Calendar.

skrustev committed:
Rename enum values containing '+' symbol.

3phase committed:
Enum values in the Badge model were capitalized, while they shouldn't be

3phase committed:
Merge branch 'design-system-model-json' of https://github.com/IgniteUI/igniteui-angular-ui-kits into design-system-model-json

wnvko committed:
chore(*): add coments to components

It is easier to follow the components with comments when code tree is collapsed.

wnvko committed:
refactor(*): update all IDs

dkamburov committed:
Merge pull request #21 from IgniteUI/mvenkov/dising-system-model-comments

Add comments to design system library

valadzhov commented:
@dkamburov  We should use either "value" or "defaultValue", throughout the file (I think).

dkamburov committed:
chore(*): Fix typo. rename `value` to `defaultValue`

3phase committed:
Adds definitions for tree grid
